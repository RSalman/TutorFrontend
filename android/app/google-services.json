{
  "project_info": {
    "project_number": "779865862095",
    "firebase_url": "https://prospr-8a50d.firebaseio.com",
    "project_id": "prospr-8a50d",
    "storage_bucket": "prospr-8a50d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:779865862095:android:1b83d4b60becf7dd",
        "android_client_info": {
          "package_name": "com.tutorfrontend"
        }
      },
      "oauth_client": [
        {
          "client_id": "779865862095-4tk4rhhj654t9tepp8c6i5elqlaj6sgg.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBDjtDNEYE5WbvPy3VumSoq1dEDhKFcOH8"
        }
      ],
      "services": {
        "analytics_service": {
          "status": 1
        },
        "appinvite_service": {
          "status": 1,
          "other_platform_oauth_client": []
        },
        "ads_service": {
          "status": 2
        }
      }
    }
  ],
  "configuration_version": "1"
}