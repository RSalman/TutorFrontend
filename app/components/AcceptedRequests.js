import React, { Component } from 'react';
import { View, StyleSheet, ListView } from 'react-native';
import { ListItem, Icon } from 'react-native-elements';
import Communications from 'react-native-communications';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import StarRating from 'react-native-star-rating';
import { updateAcceptedRequests } from '../actions/profile';
import TopNavBar from './TopNavBar';

class AcceptedRequests extends Component {
  constructor(props) {
    super(props);
    this.state = { dataSource: new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 }).cloneWithRows(this.props.acceptedRequests) };
    setInterval(() => this.props.updateAcceptedRequests(this.props.user_id), 2000);
  }

  componentDidMount() {
    this.props.updateAcceptedRequests(this.props.user_id);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ dataSource: this.state.dataSource.cloneWithRows(nextProps.acceptedRequests) });
  }

  renderRow(u) {
    return (
      <ListItem
        key={u.user_id}
        title={u.first_name + ' ' + u.last_name}
        titleStyle={style.title}
        wrapperStyle={style.listItem}
        hideChevron
        subtitle={u.course_prefix + u.course_code}
        label={
          <View>
            <StarRating
              maxStars={5}
              rating={this.props.tutorMode ? (u.agg_user_rating / u.num_user_rating) : (u.agg_tutor_rating / u.num_tutor_rating)}
              starSize={10}
              starColor={'gold'}
              selectedStar={function() {}}
              disabled
            />
            {
              // Only show button when user is in tutor mode
              this.props.tutorMode &&
                <Icon
                  name="textsms"
                  color="#29ce6b"
                  size={30}
                  containerStyle={style.icon}
                  onPress={()=> Communications.text(u.phone_number, 'Hey ' + u.first_name + '!')}
                />
            }
          </View>
        }
      />
    );
  }

  render() {
    const { approvedRequests } = this.props;
    return (
      <View style={style.container}>
        <TopNavBar toggled={this.props.toggled} tabName={this.props.tabName} />
        <ListView
          dataSource={this.state.dataSource}
          renderRow={(row) => this.renderRow(row)}
        />
      </View>
    );
  }
}
const emailIconColor = '#00aced';
const textIconColor = '#517fa4';
const white = 'white';
const blue = '#8080ff';
const style = StyleSheet.create({
  container: {
    backgroundColor: white,
    flex: 1
  },
  listItem: {
    marginRight: 20,
    marginLeft: 20,
    marginBottom:0
  },
  icon: { width: 50 },
});

const mapStateToProps = (state) => {
  return {
    user_id: state.profile.current_user.id,
    acceptedRequests: state.profile.acceptedRequests,
    tutorMode: state.profile.tutorMode,
  };
};

const mapDispatchToProps = (dispatch) => bindActionCreators({ updateAcceptedRequests }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(AcceptedRequests);
