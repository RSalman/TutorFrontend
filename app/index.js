import React, { Component } from 'react';
import { StyleSheet,TouchableHighlight, Text, View, Navigator } from 'react-native';

import HomeView from './views/HomeView';
import TutorsComponent from './components/TutorsComponent';
import LeilaFeatureView from './views/LeilaFeatureView';
import SalmanFeatureView from './views/SalmanFeatureView';
import MuraadFeatureView from './views/MuraadFeatureView';

export default class TutoringApp extends Component {
  render() {
    return (
  		<Navigator
        	initialRoute={{screen: 'HomeView'}}
        	renderScene={(route, nav) => {return this.renderScene(route, nav)}}  />
    );
  }

 renderScene(route,nav) {
    switch (route.screen) {
      	case "HomeView":
        	return <HomeView navigator={nav} />
      	case "TutorsComponent":
        	return <TutorsComponent navigator={nav} />
      	case "SalmanFeature":
        	return <SalmanFeatureView navigator={nav} /> 
        case "LeilaFeature":
        	return <LeilaFeatureView navigator={nav} /> 
        case "MuraadFeature":
        	return <MuraadFeatureView navigator={nav} /> 
      }
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
