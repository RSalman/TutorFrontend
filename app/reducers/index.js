import { combineReducers } from 'redux';
import tutors from './tutors';
import profile from './profile';

const rootReducer = combineReducers({
  tutors: tutors,
  profile: profile
});

export default rootReducer;
