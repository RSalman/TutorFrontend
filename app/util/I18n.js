import I18n from 'react-native-i18n';

import en from './locales/en';
import fr from './locales/fr';

I18n.fallbacks = true;

I18n.translations = {
  en,
  fr
};

global.I18n = I18n;
