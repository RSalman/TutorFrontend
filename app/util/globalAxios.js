import axios from 'axios';
import { Alert, NetInfo } from 'react-native';

axios.defaults.baseURL = 'http://cpserver.eastus.cloudapp.azure.com:2000/api/v1';
axios.defaults.headers.post['Content-Type'] = 'application/json';
axios.defaults.headers.get['Content-Type'] = 'application/json';
axios.defaults.headers.post.Accept = 'application/json';
axios.defaults.headers.get.Accept = 'application/json';

var connection = true;
function connectivityChange(isConnected) {
  connection = isConnected;
}

function showNoInternetDialog(config) {
  return new Promise(function(resolve, reject) {
    Alert.alert(
      I18n.t('globalWifiDialog.noInternetConnection'),
      I18n.t('globalWifiDialog.fixInternet'),
      [
        { text: I18n.t('retry'), onPress: () => resolve(axios(config)) },
      ],
      { cancelable: false }
    );
  });
}

NetInfo.isConnected.addEventListener('change', connectivityChange);

axios.interceptors.response.use(undefined, function(error) {
  return connection ? Promise.reject(error) : showNoInternetDialog(error.config);
});

global.axios = axios;
