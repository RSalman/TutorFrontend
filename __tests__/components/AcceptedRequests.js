import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import AcceptedRequests from '../../app/components/AcceptedRequests';
import '../../app/index';
import '../../app/util/globalAxios';
import '../../app/util/I18n';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

function setup(state) {
  const store = mockStore(state);
  const props = {
    updateAcceptedRequests: jest.fn(),
    store
  };
  return props;
}

describe('<AcceptedRequests />', () => {
  it('renders correctly', () => {
    const props = setup({
      rating: {
        currentlyRating: false
      },
      session: {
        tutorMode: true,
      },
      profile: {
        acceptedRequests: [],
        current_user: { id: null }
      }
    });
    const tree = renderer.create(
      <AcceptedRequests {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
