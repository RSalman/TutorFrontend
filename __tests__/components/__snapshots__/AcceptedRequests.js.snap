exports[`<AcceptedRequests /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "marginTop": 20,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "fontFamily": "HelveticaNeue-Thin",
          },
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      } />
  </View>
  <View>
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
      testID={undefined} />
    <View
      onModalHide={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "margin": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      } />
  </Modal>
</View>
`;
